src/main/java/com/taskadapter/redmineapi/internal/RedmineDateUtils.java

```diff
@@ -28,4 +33,75 @@
 	 */
 	public static final LocalDateFormat SHORT_DATE_FORMAT_V2 = new LocalDateFormat(
 			"yyyy-MM-dd");
+        
+        private static final LocalDateFormat[] dateTimeFormats = new LocalDateFormat[] {
+            FULL_DATE_FORMAT,
+            FULL_DATE_FORMAT_V2
+        };
+        
+        // SHORT_DATE_FORMAT_V2 is covered by DatatypeConverter
+        private static final LocalDateFormat[] dateFormats = new LocalDateFormat[] {
+            SHORT_DATE_FORMAT
+        };
+        
+        /**
+         * Try to parse supplied string as a datetime/date and return resulting
+         * date.
+         * 
+         * @param inputString
+         * @return parsed date
+         * @throws IllegalArgumentException in case of parse fail
+         * @throws NullPointerException if inputString is null
+         */
+        public static Date parseDate(String inputString) {
+            // The idea:
```

Comment:
I consider the iterative approach fitting - from the java side one is like the other both are mapped to dates. If something is parseable as a datetime, there is no ambiguity - at least if you asume the author did not use the strange (mm/dd/yyyy variant - yes I'm european).

This leaves only a datetime recognised as date, because the time part can't be parsed. _This_ is the reason why I prefer XML, in most cases I have an XSD and can verify against that.... To make it more concrete: Sure you can construct, but if that happens, there is more trouble. (The assumption: where the json specificiation has no valid datatype mapping, the xml definition should hold).

And: Do we really know when a short and when a long date is expected?